code/datums/records/manifest.dm

```diff
@@ -0,0 +1,206 @@
+/datum/manifest
+	/// All of the crew records.
+	var/list/general = list()
+	/// This list tracks characters spawned in the world and cannot be modified in-game. Currently referenced by respawn_character().
+	var/list/locked = list()
+	//List used entirely just for cloning.
+	var/list/cloning = list()
+	/// Total number of security rapsheet prints. Changes the header.
+	var/print_count = 0
+
+/// Builds the list of crew records for all crew members.
+/datum/manifest/proc/build()
+	for(var/i in GLOB.new_player_list)
+		var/mob/dead/new_player/readied_player = i
+		if(readied_player.new_character)
+			log_manifest(readied_player.ckey,readied_player.new_character.mind,readied_player.new_character)
+		if(ishuman(readied_player.new_character))
+			inject(readied_player.new_character)
+		CHECK_TICK
+
+/// Gets the current manifest.
+/datum/manifest/proc/get_manifest()
+	/// assoc-ing to head names, so that we give their name an officer mark on crew manifest
+	var/static/list/heads
+	if(!heads) // do not do this in pre-runtime.
+		heads = make_associative(SSdepartment.get_jobs_by_dept_id(DEPT_NAME_COMMAND))
+	/// Takes a result of each crew data in a format
+	var/list/manifest_out = list()
+
+	for(var/datum/record/crew/target in GLOB.manifest.general)
```

Comment:
nost just `target`

Reply:
I assume you meant "not just `target`", so i changed it to person_record